fix flaky test TestRetrieveValidatedConfigInfo

calling this:
```
go test ./cmd/kubeadm/app/discovery/token/... -count=1
```

sometimes panics but on a very rare occasion:

```
--- FAIL: TestRetrieveValidatedConfigInfo (0.24s)
    --- FAIL: TestRetrieveValidatedConfigInfo/valid:_retrieve_a_valid_kubeconfig_with_CA_verification_and_delayed_JWS_signature (0.24s)
        token_test.go:268: expected error false, got true, error: could not find a JWS signature in the cluster-info ConfigMap for token ID "123456"
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x16309e6]

goroutine 21 [running]:
testing.tRunner.func1.2({0x17d1600, 0x29329b0})
	/usr/local/go/src/testing/testing.go:1545 +0x238
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1548 +0x397
panic({0x17d1600?, 0x29329b0?})
	/usr/local/go/src/runtime/panic.go:914 +0x21f
k8s.io/kubernetes/cmd/kubeadm/app/discovery/token.TestRetrieveValidatedConfigInfo.func1(0xc0001e2680)
	/home/lubo/go/src/k8s.io/kubernetes/cmd/kubeadm/app/discovery/token/token_test.go:277 +0x7c6
testing.tRunner(0xc0001e2680, 0xc0003660d0)
	/usr/local/go/src/testing/testing.go:1595 +0xff
created by testing.(*T).Run in goroutine 20
	/usr/local/go/src/testing/testing.go:1648 +0x3ad
FAIL	k8s.io/kubernetes/cmd/kubeadm/app/discovery/token	0.292s
```

it panics here:
https://github.com/kubernetes/kubernetes/blob/94f15bbbcbe952762b7f5e6e3f77d86ecec7d7c2/cmd/kubeadm/app/discovery/token/token_test.go#L277
bacause of `*kubeconfig`.

to prevent the panic this:
https://github.com/kubernetes/kubernetes/blob/94f15bbbcbe952762b7f5e6e3f77d86ecec7d7c2/cmd/kubeadm/app/discovery/token/token_test.go#L271-L274

can be changed to:
```
if err != nil {
    return
}`.
```

but this fix is not enough. we have to understand why the case:
https://github.com/kubernetes/kubernetes/blob/94f15bbbcbe952762b7f5e6e3f77d86ecec7d7c2/cmd/kubeadm/app/discovery/token/token_test.go#L88
is failing.

one possibility is to increase this timeout to 1 second or more:
https://github.com/kubernetes/kubernetes/blob/94f15bbbcbe952762b7f5e6e3f77d86ecec7d7c2/cmd/kubeadm/app/discovery/token/token_test.go#L251

it could be that the go routine to apply the JWS token (60ms) completes after the timeout (200ms).

maybe, instead of trying to fix this test case we can just remove it? it tests the delayed apply of the JWS, but this is already tested in e2e:
https://testgrid.k8s.io/sig-cluster-lifecycle-kubeadm

Reproduction steps
1.Start a cpu-limited container
nerdctl run -it --rm -v /root/go/src/kubernetes:/go/kubernetes --cpus 0.2 golang:1.20.1 bash
2. Refer to [https://gist.github.com/liggitt/6a3a2217fa5f846b52519acfc0ffece0#running-unit-tests-to-reproduce-flakes](https://gist.github.com/liggitt/6a3a2217fa5f846b52519acfc0ffece0#running-unit-tests-to-reproduce-flakes.) for stressthe ut https://gist.github.com/liggitt/6a3a2217fa5f846b52519acfc0ffece0#running-unit-tests-to-reproduce-flakes.
/go/bin/stress ./token.test -test.run TestRetrieveValidatedConfigInfo
3. The result
/go/bin/stress ./token.test -test.run TestRetrieveValidatedConfigInfo
...
25s: 65 runs so far, 0 failures

/tmp/go-stress-20231120T151311-2381021825
--- FAIL: TestRetrieveValidatedConfigInfo (0.21s)
    --- FAIL: TestRetrieveValidatedConfigInfo/valid:_retrieve_a_valid_kubeconfig_with_CA_verification_and_delayed_JWS_signature (0.21s)
        token_test.go:268: expected error false, got true, error: could not find a JWS signature in the cluster-info ConfigMap for token ID "123456"
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x23313f5]

goroutine 26 [running]:
testing.tRunner.func1.2({0x24df860, 0x3791190})
	/usr/local/go/src/testing/testing.go:1526 +0x372
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1529 +0x650
panic({0x24df860, 0x3791190})
	/usr/local/go/src/runtime/panic.go:890 +0x263
k8s.io/kubernetes/cmd/kubeadm/app/discovery/token.TestRetrieveValidatedConfigInfo.func1(0xc0004f8ea0)
	/root/go/src/kubernetes/cmd/kubeadm/app/discovery/token/token_test.go:277 +0xf15
testing.tRunner(0xc0004f8ea0, 0xc0003c2b90)
	/usr/local/go/src/testing/testing.go:1576 +0x217
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1629 +0x806


ERROR: exit status 2

....
1m30s: 237 runs so far, 1 failures (0.42%)

/tmp/go-stress-20231120T151311-486950637
--- FAIL: TestRetrieveValidatedConfigInfo (0.21s)
    --- FAIL: TestRetrieveValidatedConfigInfo/valid:_retrieve_a_valid_kubeconfig_with_CA_verification_and_delayed_JWS_signature (0.21s)
        token_test.go:268: expected error false, got true, error: could not find a JWS signature in the cluster-info ConfigMap for token ID "123456"
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x23313f5]

goroutine 37 [running]:
testing.tRunner.func1.2({0x24df860, 0x3791190})
	/usr/local/go/src/testing/testing.go:1526 +0x372
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1529 +0x650
panic({0x24df860, 0x3791190})
	/usr/local/go/src/runtime/panic.go:890 +0x263
k8s.io/kubernetes/cmd/kubeadm/app/discovery/token.TestRetrieveValidatedConfigInfo.func1(0xc000500000)
	/root/go/src/kubernetes/cmd/kubeadm/app/discovery/token/token_test.go:277 +0xf15
testing.tRunner(0xc000500000, 0xc0004f80c0)
	/usr/local/go/src/testing/testing.go:1576 +0x217
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1629 +0x806


ERROR: exit status 2

...
2m5s: 329 runs so far, 2 failures (0.61%)

/tmp/go-stress-20231120T151311-2897237707
--- FAIL: TestRetrieveValidatedConfigInfo (0.21s)
    --- FAIL: TestRetrieveValidatedConfigInfo/valid:_retrieve_a_valid_kubeconfig_with_CA_verification_and_delayed_JWS_signature (0.21s)
        token_test.go:268: expected error false, got true, error: could not find a JWS signature in the cluster-info ConfigMap for token ID "123456"
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x23313f5]

goroutine 25 [running]:
testing.tRunner.func1.2({0x24df860, 0x3791190})
	/usr/local/go/src/testing/testing.go:1526 +0x372
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1529 +0x650
panic({0x24df860, 0x3791190})
	/usr/local/go/src/runtime/panic.go:890 +0x263
k8s.io/kubernetes/cmd/kubeadm/app/discovery/token.TestRetrieveValidatedConfigInfo.func1(0xc000509520)
	/root/go/src/kubernetes/cmd/kubeadm/app/discovery/token/token_test.go:277 +0xf15
testing.tRunner(0xc000509520, 0xc0000732d0)
	/usr/local/go/src/testing/testing.go:1576 +0x217
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1629 +0x806


ERROR: exit status 2

...


When  increase this timeout to 500ms:https://github.com/kubernetes/kubernetes/blob/94f15bbbcbe952762b7f5e6e3f77d86ecec7d7c2/cmd/kubeadm/app/discovery/token/token_test.go#L251 
The result has no failed:
root@b3f1563fa4d8:/go/kubernetes# /go/bin/stress ./token.test -test.run TestRetrieveValidatedConfigInfo
...
7m45s: 1247 runs so far, 0 failures
7m50s: 1260 runs so far, 0 failures
...

--- a/cmd/kubeadm/app/discovery/token/token_test.go
+++ b/cmd/kubeadm/app/discovery/token/token_test.go
@@ -248,7 +248,7 @@ users: null
 			}
 
 			// Set arbitrary discovery timeout and retry interval
-			test.cfg.Timeout = &metav1.Duration{Duration: time.Millisecond * 200}
+			test.cfg.Timeout = &metav1.Duration{Duration: time.Millisecond * 500}
 			interval := time.Millisecond * 20
 
 			// Patch the JWS signature after a short delay

